sparta/sparta/kernel/Scheduler.hpp

```diff
@@ -232,12 +263,26 @@ class Scheduler : public RootTreeNode
         void addEvent(uint32_t firing_group, Scheduleable * scheduleable) {
             sparta_assert(firing_group > 0);
             sparta_assert(firing_group < groups.size());
-            groups[firing_group].emplace_back(scheduleable);
+            groups[firing_group].addScheduleable(scheduleable);
             first_group_idx = std::min(first_group_idx, firing_group);
         }
 
-        Tick tick;     //!< The tick this quantum represents
-        Groups groups; //!< The list of firing groups. This is indexed by dag_group+1
+        void addEventIfNotScheduled(uint32_t firing_group, Scheduleable * scheduleable) {
+            sparta_assert(firing_group > 0);
+            sparta_assert(firing_group < groups.size());
+            auto & grp = groups[firing_group];
+            const auto grp_size = grp.size();
+            for(uint32_t idx = 0; idx < grp_size; ++idx) {
```

Comment:
Could you use `std::find_if` instead?

Reply:
I could, but that'd add a second `if` statement.  Currently if there's a match, we bail out of this function right away.  If I change it to `find_if`, then I'd have to see if there were a match after the alg did all of the first `if` statements.  Good suggestion through, that'd clean it up.